packages/sdk/contractkit/src/wrappers/SortedOracles.test.ts

```diff
@@ -35,252 +40,432 @@ testWithGanache('SortedOracles Wrapper', (web3) => {
+    target: ReportTarget,
+    expiredOracles: Address[],
+    allOracles: Address[]
+  ): Promise<void> {
     const expirySeconds = (await sortedOracles.reportExpirySeconds()).toNumber()
-    await reportAsOracles(expiredOracles)
+    await reportAsOracles(sortedOracles, target, expiredOracles)
     await timeTravel(expirySeconds + 5, web3)
-    const freshOracles = stableTokenOracles.filter((o) => !expiredOracles.includes(o))
-    await reportAsOracles(freshOracles)
+    const freshOracles = allOracles.filter((o) => !expiredOracles.includes(o))
+    await reportAsOracles(sortedOracles, target, freshOracles)
   }
 
-  beforeAll(async () => {
-    sortedOracles = await kit.contracts.getSortedOracles()
-    stableTokenAddress = await kit.registry.addressFor(CeloContract.StableToken)
-    allAccounts = await web3.eth.getAccounts()
-    nonOracleAddress = allAccounts.find((addr) => {
-      return !stableTokenOracles.includes(addr)
-    })!
-  })
+  /**
+   * When testing with a custom token pair we can't use the
+   * already deployed SortedOracles because that's managed
+   * by Governance and we can't execute changes on the contract.
+   * To make it easier we'll deploy an new version for use in
+   * the tests
+   */
+  async function newSortedOracles(owner: Address): Promise<SortedOraclesWrapper> {
+    const instance = await SortedOracles.new({ from: owner })
+    await instance.initialize(NetworkConfig.oracles.reportExpiry, { from: owner })
+    return new SortedOraclesWrapper(kit, instance.contract)
+  }
 
-  describe('#report', () => {
-    const value = 16
+  async function addOracleForPair(
+    sortedOraclesInstance: SortedOraclesWrapper,
+    pair: PairWithIdentifier,
+    oracle: Address,
+    owner: Address
+  ): Promise<void> {
+    // @ts-ignore
+    const sortedOraclesContract = sortedOraclesInstance.contract
+    await sortedOraclesContract.methods.addOracle(pair.identifier, oracle).send({
+      from: owner,
+    })
+  }
 
-    describe('when reporting from a whitelisted Oracle', () => {
-      it('should be able to report a rate', async () => {
-        const initialRates: OracleRate[] = await sortedOracles.getRates(CeloContract.StableToken)
+  // NOTE: These values are set in test-utils/network-config.json, and are derived
+  // from the MNEMONIC. If the MNEMONIC has changed, these will need to be reset.
+  // To do that, look at the output of web3.eth.getAccounts(), and pick a few
+  // addresses from that set to be oracles
+  const stableTokenOracles: Address[] = NetworkConfig.stableToken.oracles
+  // Use same oracle addresses for CELO/BTC as well
+  const celoBtcOracles: Address[] = NetworkConfig.stableToken.oracles
+  const oracleAddress = stableTokenOracles[stableTokenOracles.length - 1]
 
-        const tx = await sortedOracles.report(CeloContract.StableToken, value, oracleAddress)
-        await tx.sendAndWaitForReceipt()
+  const celoBtcPair = newPairWithIdentifier('CELO/BTC')
```

Comment:
Probably doesn't make sense to do now but when the CELO/BTC rate is live it would be cool to have `CELO/BTC` exist in an enum in contractkit

Reply:
Resolved by the refactor